src/pstl_offload.cpp

```diff
@@ -200,22 +393,162 @@ __internal_msize(void* __user_ptr)
 #endif
     }
 
-    std::size_t __res = 0;
     __block_header* __header = static_cast<__block_header*>(__user_ptr) - 1;
 
     if (__same_memory_page(__user_ptr, __header) && __header->_M_uniq_const == __uniq_type_const)
     {
-        __res = __header->_M_requested_number_of_bytes;
+        return __header->_M_requested_number_of_bytes;
+    }
+
+    std::optional<std::size_t> __size = __large_aligned_ptrs_map::__get_size(__large_aligned_ptrs, __user_ptr);
+
+    return __size.has_value() ? __size.value() : __original_msize(__user_ptr);
```

Comment:
```suggestion
    return __size.has_value() ? *__size : __original_msize(__user_ptr);
```
Since we already checked if `optional` has a value it's better to call the API that doesn't throw.

Reply:
Done.